events/tests/test_models.py

```diff
@@ -186,3 +185,25 @@ def test_event_previous_event(self):
         # 'Event.previous_event' can return None if there is no
         # OccurringRule or RecurringRule found.
         self.assertIsNone(self.event.previous_event)
+
+    def test_scheduled_to_start_this_year_method(self):
+        now = seconds_resolution(timezone.now())
+
+        occurring_time_dtstart = now + datetime.timedelta(days=3)
```

Comment:
Hmmmm... Thinking about this test again. On the off chance that tests are run between the 29th of December and the 31st of December, this test will fail since the next year will be set to 3 days ahead...

So if tests are run on the 29th of December, `occurring_time_dtstart` will be set to 3 days from then which would be the 1st of January (the following year)

Reply:
You can use something like https://github.com/spulec/freezegun to freeze time so the tests run reliably.

Reply:
> You can use something like https://github.com/spulec/freezegun to freeze time so the tests run reliably.

Thanks, @hugovk, for suggesting [freezegun](https://github.com/spulec/freezegun) to handle the time-dependent tests. I have tested the updates locally, and I’m ready to push the changes to the current pull request. However, before proceeding, I wanted to get clarity on one remaining concern regarding the introduction of external dependencies into the project.

Specifically, I’m unsure of the current team policy regarding the addition of new dependencies. My concern revolves around potential risks that come with relying on third-party libraries, such as maintenance issues, security vulnerabilities, and compatibility with future versions of Python/Django.

In this case, while freezegun is a robust library for freezing time, I wonder if it’s necessary to introduce an entire dependency solely to address the time-delta requirements of the Event object in the tests. I'm considering whether there might be an alternative approach that avoids the need for an external library, such as directly mocking time or using built-in Python features to clamp dates.

I would appreciate your thoughts on whether the use of freezegun is acceptable for this situation, or if the team would prefer exploring another solution.

Looking forward to your feedback,

Reply:
Thanks for your question, it's always good to keep these things in mind. My feeling is that it would be okay, as there are already a number of third-party dependencies, but I'll leave it to the maintainers to answer more concretely.

An alternative would be to use something like `unitest.mock` to fake the date returned by `datetime.timedelta`.

Reply:
Good day @hugovk and all,

I've opted to mock timezone (https://github.com/alvindera97/pythondotorg/blame/c4db8d42591f359eb899a5940ea25b71aebcf8a3/events/tests/test_models.py#L191-L247). While Django's import path for timezone seems stable, any future changes would require updating the mock path. Currently, I'm not aware of any deprecations or changes affecting timezone.